api/server/router/grpc/grpc.go

```diff
@@ -2,12 +2,17 @@ package grpc // import "github.com/docker/docker/api/server/router/grpc"
 
 import (
 	"context"
+	"fmt"
+	"os"
 	"strings"
 
 	"github.com/docker/docker/api/server/router"
-	grpc_middleware "github.com/grpc-ecosystem/go-grpc-middleware"
 	"github.com/moby/buildkit/util/grpcerrors"
+	"github.com/moby/buildkit/util/stack"
+	"github.com/moby/buildkit/util/tracing"
```

Comment:
We were already depending on these, correct? (just making sure we want to couple _more_ or _less_ on BuildKit here)

I don't think any of this would end up on the client-side, but just in case, we should also do a test "vendor" PR from this branch in docker/cli to make sure we don't reintroduce BuildKit as a dependency there.

Reply:
For sure. This is still drafted, it was just a first pass, so I'll make sure of that before I mark the PR as ready.

Reply:
Yeah, I should've waited until no longer draft; just peeking (happy to see you're working on this!)

Reply:
We already depended on these, and I don't think these changes should end up in the CLI due to where they happened, but can do a test vendor PR there to confirm.